onmt/translate/translator.py

```diff
@@ -1091,6 +1091,9 @@ def _translate_batch_with_strategy(
             )
 
             if step == 0:
+                log_probs = fn_map_state(log_probs, dim=1)
+                if fn_map_state is not None:
```

Comment:
This does not look good.

Reply:
Can you be more precise ?

Reply:
You could not use the variable as a function and then check if it's None or not. Also, I see there was another call for the same process a few lines before which seems redundant.
But now, it's all good.